arch/arm/Kconfig

```diff
@@ -901,6 +901,13 @@ config ARM_HAVE_NEON
 	---help---
 		Decide whether support NEON instruction
 
+config ARM_FP_ABI_SOFT
```

Comment:
why need ARM_FP_ABI_SOFT? If the hardware has FPU, it's better to enable hard, otherwise it must be softfp.

Reply:
This may be relevant if the application is compiled with soft and the kernel may use hard. I mean if there are both app and kernel using soft and then kernel is upgraded to a newer version and enable hard, but there is a need to keep same binary for app. Does this makes sense to you?

Reply:
I still check. If none of syscalls use neither float nor double then we can remove softfp option

Reply:
Also softfp allows to use hard float with toolchain based on GCC 9 and lower.

Reply:
Emm, but it's a bug of old toolchain, I think in this case we should suggest user to update their toolchain instead of do some magic in correct code to make toolchain happy.

And softfp is a valid option (disabled by default), it can make Toolchain.defs more general (if some one want to use softfp, just enable this option instead of modifiy the board level Make.defs)

Reply:
@no1wudi could you try to build some board with fpu and softfp and disable CONFIG_LIBM? I am curious whether the toolchain provide the built-in library(e.g. libgcc.a, libm.a, libsupc++...) with this combination.

Reply:
It works on sabre-6quad and giantboard, and see:
```
➜  nuttx git:(rv) ls /usr/lib/gcc/arm-none-eabi/10.3.1/thumb/
nofp/             v7/               v7-a+fp/          v7e-m/            v7e-m+fp/         v7-m/             v7ve+simd/        v8-a/             v8-m.base/        v8-m.main+dp/                                                     
v6-m/             v7-a/             v7-a+simd/        v7e-m+dp/         v7+fp/            v7-r+fp.sp/       v8.1-m.main+mve/  v8-a+simd/        v8-m.main/        v8-m.main+fp/ 
```

Reply:
Thanks for test.

Reply:
ARM_FP_ABI_SOFT->ARM_FPU_ABI_SOFT?

Reply:
Done

Reply:
> Emm, but it's a bug of old toolchain, I think in this case we should suggest user to update their toolchain instead of do some magic in correct code to make toolchain happy.
> 
> And softfp is a valid option (disabled by default), it can make Toolchain.defs more general (if some one want to use softfp, just enable this option instead of modifiy the board level Make.defs)

From my practice toolchain update is not always a straight forward process and migration to a new toolchain take a lot of time and efforts. So I'm fine if we leave `softfp` as a fallback option for those who still are not ready to change toolchain withing a single day.